docs/configure.md

```diff
@@ -0,0 +1,104 @@
+# CMake's configuration process
+
+CMake Tools wraps the CMake configure process separately from the build process.
+
+## CMake configuration process overview
+
+In CMake, _configure_ refers to detecting requirements and generating the build files that will produce the final compiled artifacts.
+
+The following concepts will help you understand how CMake Tools interacts with CMake's configure process:
+
+* The _CMake Cache_ is a list of key-value pairs that persist between runs of the configure process. It contains the following:
+
+    * Values that are expensive to determine, such as whether a `-flag` or `#include` file is supported by the compiler.
+    * Values that rarely change, such as the path to a header/library.
+    * Values that offer control to the developer, such as `BUILD_TESTING` to determine whether or not to build test libraries/executables.
+* _Cache initializer arguments_ are the arguments passed to CMake that set values in the cache before any CMake scripts are run. These allow you to control the build settings. On the CMake command line, these appear as `-D` arguments. (CMake Tools doesn't use CMake's `-C` argument).
+* Unless overwritten or deleted, values in the CMake Cache persist between CMake runs.
+* CMake doesn't do the build itself, it relies on build tools installed on your system. The result of a _configure_ depends on the CMake _Generator_. The _Generator_ tells CMake what kind of tool will be used to compile and generate the results of the build. There are several families of generators available:
+
+    |Generator |Description|
+    |---------|---------|
+    |Ninja | Emits files for the [Ninja build tool](https://ninja-build.org). This is the generator CMake Tools tries first, unless configured otherwise. See [cmake.preferredGenerators](cmake-settings.md#cmake-settings). |
+    |Makefile |  Emits a `Makefile` for the project that can be built via `make`.|
+    |Visual Studio     | Emits visual studio solutions and project files. There are many different Visual Studio generators, so it is recommended to let CMake Tools automatically determine the appropriate generator.|
+
+Regardless of generator, CMake Tools always supports building from within Visual Studio Code. If you are building from within Visual Studio Code, we recommend you use the [Ninja build tool](https://ninja-build.org/).
+
+## How CMake Tools configures
+
+CMake Tools drives CMake via CMake Server, an execution mode of CMake that provides a persistent connection to get project information.
```

Comment:
Since we document what we currently have, rather than the history of where we've been, I changed this to say that it uses the cmake-file-api and dropped mention of cmake server.

There was another reference to the cmake server in this topic, which I also changed. It talked about setting environment variables for the cmake server process. I changed it like so: "CMake Tools sets environment variables necessary to use the selected Visual Studio installation, and sets `CC` and `CXX` to `cl.exe` so that CMake will detect the Visual C++ compiler as the primary compiler, even if other compilers like GCC are present on `$Path`."